<launch>
    <!--name of gazebo launch file, without '_world.launch'-->
    <arg name="world" default="lane"/>
    <arg name="blank" default="false"/>>
    <arg name="viz_type" default="local"/>
    <arg name="run_as_remote" default="false"/>
    <arg name="run_client" default="true"/>
    <arg name="timeout" default="-1"/>
    <arg name="record" default="false"/>
    <arg name="bag_prefix" default="lane_demo_sim"/>
    <arg name="topics" default="-a" />

    <param name="client_comm_type" value="local"/>
    <param name="client_context_type" value="sim"/>
    <include file="$(find auto_rc_car_worlds)/launch/$(arg world)_world.launch">
        <arg name="blank" value="$(arg blank)"/>
    </include>


    <include file="$(find auto_rc_car_description)/launch/viz_car_$(arg viz_type).launch"/>
    

    <group if="$(arg run_as_remote)">
        <node name="core" pkg="auto_rc_car_api" type="remote_core_node.py" output="screen" />
        <param name="client_comm_type" value="remote"/>
    </group>
    
    <group unless="$(arg run_as_remote)">
        <param name="client_comm_type" value="local"/>
    </group>

    <!--
    <group if="$(arg run_client)">
        <node name="client" pkg="auto_rc_car_demos" type="simple_lane_drive.py" output="screen"/>
    </group>
-->
    <group if="$(arg record)">
        <node pkg="rosbag" type="record" name="bag_logger"
            args="record -o /home/$(env USER)/$(arg bag_prefix) $(arg topics)"/>
    </group>

</launch>
